test/TorchSharpTest/TestTorchVisionUtils.cs

```diff
@@ -1,18 +1,42 @@
+using System;
 using System.IO;
 using Xunit;
 
 namespace TorchSharp
 {
     public class TestTorchVisionUtils
     {
+        private class MockImager : torchvision.io.Imager
+        {
+#region " Mock implementation "
+            public override torch.Tensor DecodeImage(Stream stream, torchvision.io.ImageReadMode mode = torchvision.io.ImageReadMode.UNCHANGED)
+            {
+                throw new NotImplementedException();
+            }
+
+            public override torch.Tensor DecodeImage(byte[] data, torchvision.io.ImageReadMode mode = torchvision.io.ImageReadMode.UNCHANGED)
+            {
+                throw new NotImplementedException();
+            }
+
+            public override void EncodeImage(torch.Tensor image, torchvision.ImageFormat format, Stream stream)
+            {
+            }
+
+            public override byte[] EncodeImage(torch.Tensor image, torchvision.ImageFormat format)
+            {
+                throw new NotImplementedException();
+            }
+#endregion
+        }
+
         [Fact]
         public void Save_Image_TestMemoryUsage()
         {
-            var imager = new torchvision.io.SkiaImager();
-            using var ms = new MemoryStream();
+            var imager = new MockImager();
             using var image = torch.randn(32, 3, 32, 32);
             using (var d = torch.NewDisposeScope()) {
-                torchvision.utils.save_image(image, ms, torchvision.ImageFormat.Png, imager: imager);
+                torchvision.utils.save_image(image, (Stream)null, torchvision.ImageFormat.Png, imager: imager);
```

Comment:
Explain, please. How does this address the intermittent errors?

Reply:
Well I was going to mock the memory stream too, as the test doesn't require it, but mocking the Imager means the memorystream isn't used.

I can put the MemoryStream back, the real fix is mocking the Imager.

Reply:
I was more thinking about the intermittency (is that a word??) of the failures -- they happen all over the place, and fairly randomly. I believe that it's because unit tests are run concurrently, and there are probably non-reentrancy issues in the native code that have yet to be identified. Therefore, the fact that it failed in your new unit test is probably a pure coincidence.

Reply:
Either way, my unit test is not intended to check whether SkiaImager works reliably. I use Mocks to limit the scope of the test, and I find it gives better granularity that way.